worker/buildbot_worker/test/util/misc.py

```diff
@@ -186,7 +188,16 @@ class StdoutAssertionsMixin(object):
     """
 
     def setUpStdoutAssertions(self):
-        self.stdout = cStringIO.StringIO()
+        #
+        # sys.stdout is implemented differently
+        # in Python 2 and Python 3, so we need to
+        # override it differently.
+        # In Python 2, sys.stdout is a byte stream.
+        # In Python 3, sys.stdout is a text stream.
+        if PY3:
```

Comment:
I'd expect this kind of thing to be handled by `future`, does it not handle it well?